robottelo/install/base.py

```diff
@@ -0,0 +1,152 @@
+# -*- encoding: utf-8 -*-
+"""Generic base class for install commands."""
+import logging
+
+from robottelo import ssh
+from robottelo.config import conf
+
+
+class InstallError(Exception):
+    """Indicates that a install command could not be run."""
+
+
+class Base(object):
+    """
+    @param command_base: base install command.
+    """
+    command_base = None  # each inherited instance should define this
+    command_requires_auth = False  # True when command requires auth user
+    command_service = None
+
+    logger = logging.getLogger('robottelo')
+
+    @classmethod
+    def reset(cls, options=None):
+        """
+        dummy
+        """
+
+        if options is None:
+            options = {}
+
+        result = cls.execute(cls._construct_command(options))
+        return result
+
+    @classmethod
+    def execute(cls, command, user=None, password=None, timeout=None):
+        """Executes  command on the server via ssh"""
+        user, password = cls._get_username_password(user, password)
+
+        # add time to measure hammer performance
+        perf_test = conf.properties.get('performance.test.foreman.perf', '0')
+        cmd = u'LANG={0} {1} {2}'.format(
+            conf.properties['main.locale'],
+            u'time -p' if perf_test == '1' else '',
+            command,
+        )
+
+        return ssh.command(cmd.encode('utf-8'), timeout=timeout)
+
+    @classmethod
+    def service_restart(cls, options=None):
+        """
```

Comment:
Single line docstrings should follow the format:

```python
"""Single line comment."""
```

On the other hand mult line comments should follow the format:

```python
"""First line should be in front of opening docstring

Second line....
"""
```